src/datanomy/tui/parquet.py

```diff
@@ -782,80 +784,96 @@ def _read_data(self) -> tuple[Any, int, int]:
 
         return table, num_rows_display, total_rows
 
-    def _create_data_table(self, table: Any, num_rows_display: int) -> Panel:
+    def _create_data_table(self, table: Any, num_rows_display: int) -> DataTable:
         """
-        Create Rich table with data rows.
+        Create a Textual ``DataTable`` widget populated with preview rows.
 
         Parameters
         ----------
-            table: PyArrow table
-            num_rows_display: Number of rows to display
+            table: PyArrow table slice for preview rendering
+            num_rows_display: Number of rows to include in the table
 
         Returns
         -------
-            Panel: Rich Panel containing data table
-        """
-        # Create Rich table
-        data_table = Table(
-            show_header=True,
-            header_style="bold cyan",
-            border_style="dim",
-            expand=False,
-            box=None,
-        )
+            DataTable: Configured widget ready for display
+        """
+
+        data_table: DataTable = DataTable(id="data-preview-table")
+        data_table.show_row_labels = False
+        data_table.show_cursor = True
+        data_table.cursor_type = "cell"
+        data_table.zebra_stripes = True
+        data_table.border_title = "Data Preview"
+        data_table.styles.border = ("round", "cyan")
+        data_table.styles.width = "auto"
+        data_table.styles.margin = (1, 0, 0, 0)
```

Comment:
Some of those seem to be just the default, unnecessary or can be used on the class constructor.
```suggestion
        data_table: DataTable = DataTable(id="data-preview-table", zebra_stripes=True)
        data_table.border_title = "Data Preview"
        data_table.styles.border = ("round", "cyan")
        data_table.styles.width = "auto"
```